### PR TITLE
fix: LRUcache for block requests

### PR DESCRIPTION
**Pull request by TarikGul**
closes: [#612](https://github.com/paritytech/substrate-api-sidecar/issues/612)

LRUcache implemented for a `/blocks/head` & `/blocks/{blockId}`.

The main benefit of having this LRUCache is making sure that when users are polling `/blocks/head` for the most recent block it doesnt make repeat requests for the same blockHash.

### CODE REVIEW

**Review by insipx — APPROVED** [review]
LGTM!

File: src/chains-config/cache/lruCache.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import LRU from 'lru-cache';
+
+import { IBlock } from '../../types/responses';
+
+export const initLRUCache = (): LRU<string, IBlock> => {
+	const config = { max: 2 };
```
Comment (niklasad1) [inline src/chains-config/cache/lruCache.ts:6]: what does `max == 2` here? Max two blocks can be cached in memory?

Comment (TarikGul) [reply src/chains-config/cache/lruCache.ts:6]: Yup, exactly. The max length of the cache wont exceed 2. This is actually kinda up in the air, but from a brief discussion with Zeke we figured 1-2 would be more sufficient for the needs of polling `blocks/head`.

**Review by niklasad1 — APPROVED** [review]

**Comment by TarikGul** [comment]
Before merging this too, I am going to confirm that when running stress tests, and benchmarks that this doesnt cause any residual issues for sidecar.

**Comment by TarikGul** [comment]
After running benchmarks against 9.1.4 and this PR, there were no regressions:

## v9.1.4

```
Running 2m test @ http://127.0.0.1:8080
  4 threads and 12 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency     5.37s     2.49s   10.96s    67.65%
    Req/Sec     1.97      4.97    30.00     89.19%
  Latency Distribution
     50%    5.19s 
     75%    7.43s 
     90%    8.42s 
     99%   10.96s 
  267 requests in 2.00m, 292.36MB read
Requests/sec:      2.22
Transfer/sec:      2.43MB

Total completed requests: 267
Failed requests: 0
Timeouts: 0
Avg RequestTime(Latency):          5371.86ms
Max RequestTime(Latency):          10958.036ms
Min RequestTime(Latency):          188.677ms
```

## This branch

```
Running 2m test @ http://127.0.0.1:8080
  4 threads and 12 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency     5.21s     2.83s   13.46s    76.29%
    Req/Sec     2.62      5.79    30.00     85.00%
  Latency Distribution
     50%    4.64s 
     75%    7.65s 
     90%    8.13s 
     99%   13.46s 
  279 requests in 2.00m, 308.50MB read
Requests/sec:      2.32
Transfer/sec:      2.57MB

Total completed requests: 279
Failed requests: 0
Timeouts: 0
Avg RequestTime(Latency):          5206.54ms
Max RequestTime(Latency):          13458.607ms
Min RequestTime(Latency):          12.891ms
```

No regressions I can tell, ran two more sets of benchmarks and they all are similar. 

@dvdplm any last thoughts before I merge this?

**Pull request merged**